migrations/4_genesis.js

```diff
@@ -6,28 +6,59 @@ const Controller = artifacts.require("Controller")
 const ManagerProxy = artifacts.require("ManagerProxy")
 const LivepeerToken = artifacts.require("LivepeerToken")
 const GenesisManager = artifacts.require("GenesisManager")
-const TokenDistributionMock = artifacts.require("TokenDistributionMock")
+const MerkleMine = artifacts.require("MerkleMine")
+
+const getCurrentBlock = async () => {
+    return await new Promise((resolve, reject) => {
+        return web3.eth.getBlock("latest", (err, blk) => {
+            if (err) {
+                reject(err)
+            } else {
+                resolve(blk)
+            }
+        })
+    })
+}
 
 module.exports = function(deployer, network, accounts) {
     deployer.then(async () => {
         const lpDeployer = new ContractDeployer(deployer, Controller, ManagerProxy)
 
         const controller = await Controller.deployed()
-
         const tokenAddr = await controller.getContract(contractId("LivepeerToken"))
         const minterAddr = await controller.getContract(contractId("Minter"))
-
         const token = await LivepeerToken.at(tokenAddr)
 
-        const dummyTokenDistribution = await lpDeployer.deploy(TokenDistributionMock, genesis.dummyTokenDistribution.timeToEnd)
-
         let genesisManager
 
+        const grantsStartTimestamp = (await getCurrentBlock()).timestamp + genesis.timeToGrantsStart
+
         if (!lpDeployer.isProduction(network)) {
-            // If not in production, send the company supply to the deployment account instead of the bank multisig
-            genesisManager = await lpDeployer.deploy(GenesisManager, tokenAddr, dummyTokenDistribution.address, accounts[0], minterAddr)
+            // If not in production, send the crowd supply to the faucet and the company supply to the deployment account
+            deployer.logger.log("Not in production - crowd supply will be sent to faucet and company supply will be sent to deployment account")
+
+            const faucetAddr = await controller.getContract(contractId("LivepeerTokenFaucet"))
+            genesisManager = await lpDeployer.deploy(GenesisManager, tokenAddr, faucetAddr, accounts[0], minterAddr, grantsStartTimestamp)
         } else {
-            genesisManager = await lpDeployer.deploy(GenesisManager, tokenAddr, dummyTokenDistribution.address, genesis.bankMultisig, minterAddr)
+            // If in production, send the crowd supply to the token distribution contract and the company supply to the bank multisig
+            deployer.logger.log("In production - crowd supply will be sent to token distribution contract and company supply will be sent to bank multisig")
+
+            const currentBlockNum = (await getCurrentBlock()).number
+            const callerAllocationStartBlock = currentBlockNum + genesis.merkleMine.blocksToCliff
```

Comment:
Instead of hardcoding values for `callerAllocationStartBlock` and `callerAllocationEndBlock` in `migrations/genesis.config.js`, we can hardcode the desired values for `blocksToCliff` and `callerAllocationPeriod` in that file. As a result, we can calculate the desired values for `callerAllocationStartBlock` and `callerAllocationEndBlock` when we actually deploy the MerkleMine contract instead of setting those values at the very beginning of the migration scripts which can create weird scenarios where for example we set `callerAllocationStartBlock` to 50 when the current block is 25, but by the time the migration scripts are deploying the MerkleMine contract the current block is already at 51.